tests/unit/test_data_interface.py

```diff
@@ -0,0 +1,70 @@
+"""
+Tests the data interfacing layer for various objects
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import unittest
+
+import ga4gh.client as client
+import ga4gh.datarepo as datarepo
+import ga4gh.backend as backend
+import ga4gh.protocol as protocol
+import tests.paths as paths
+
+
+class TestInterfacingLayer(unittest.TestCase):
+
+    def _assertValid(self, obj, clazz):
+        jsonString = protocol.toJson(obj)
+        isValid = protocol.validate(jsonString, clazz)
+        if not isValid:
+            raise Exception(obj)
+
+    def _getPageSizes(self, num):
+        return [None, 1, 2, 3, num - 1, num, num + 1]
+
+    def setUp(self):
+        self._repo = datarepo.SqlDataRepository(paths.testDataRepo)
+        self._repo.open(datarepo.MODE_READ)
+        self._backend = backend.Backend(self._repo)
+        self._client = client.LocalClient(self._backend)
+
+    def testGetReferenceSets(self):
+        repoReferenceSets = self._repo.getReferenceSets()
+        for repoReferenceSet in repoReferenceSets:
+            referenceSetId = repoReferenceSet.getId()
+            referenceSet = self._client.getReferenceSet(referenceSetId)
+            self._assertValid(referenceSet, protocol.ReferenceSet)
```

Comment:
Also a deep compare would seem to require snake caseing all the attributes... otherwise how to compare a ncbi_taxon_id to a ncbiTaxonId?  Are we ready to do that?  Should we do that?